### PR TITLE
Move support libraries to androidx.

### PR DESCRIPTION
**Pull request by NightlyNexus**

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -6,18 +6,18 @@ buildscript {
...
   ext.deps = [
     androidPlugin: 'com.android.tools.build:gradle:3.2.0-beta02',
     okhttp: "com.squareup.okhttp3:okhttp:${versions.okhttp}",
     mockWebServer: "com.squareup.okhttp3:mockwebserver:${versions.okhttp}",
     pollexor: 'com.squareup:pollexor:2.0.4',
-    supportCompat: "com.android.support:support-compat:${versions.supportLibrary}",
-    supportFragment: "com.android.support:support-fragment:${versions.supportLibrary}",
-    supportAnnotations: "com.android.support:support-annotations:${versions.supportLibrary}",
-    supportExifInterface: "com.android.support:exifinterface:${versions.supportLibrary}",
+    supportCompat: "androidx.appcompat:appcompat:${versions.supportLibrary}",
```
Comment (JakeWharton) [inline build.gradle]: Wrong dep. Were you looking for androidx.core:core?

File: build.gradle [hunk 2]
```diff
@@ -6,18 +6,18 @@ buildscript {
     'sourceCompatibility': JavaVersion.VERSION_1_7,
     'targetCompatibility': JavaVersion.VERSION_1_7,
     'okhttp': '3.10.0',
-    'supportLibrary': '28.0.0-alpha3'
+    'supportLibrary': '1.0.0-rc01'
```
Comment (JakeWharton) [inline build.gradle]: We can't merge until 1.0 final.

File: build.gradle [hunk 3]
```diff
@@ -6,18 +6,18 @@ buildscript {
...
     androidPlugin: 'com.android.tools.build:gradle:3.2.0-beta02',
     okhttp: "com.squareup.okhttp3:okhttp:${versions.okhttp}",
     mockWebServer: "com.squareup.okhttp3:mockwebserver:${versions.okhttp}",
     pollexor: 'com.squareup:pollexor:2.0.4',
-    supportCompat: "com.android.support:support-compat:${versions.supportLibrary}",
-    supportFragment: "com.android.support:support-fragment:${versions.supportLibrary}",
-    supportAnnotations: "com.android.support:support-annotations:${versions.supportLibrary}",
-    supportExifInterface: "com.android.support:exifinterface:${versions.supportLibrary}",
+    supportCompat: "androidx.appcompat:appcompat:${versions.supportLibrary}",
+    supportFragment: "androidx.fragment:fragment:${versions.supportLibrary}",
```
Comment (JakeWharton) [inline build.gradle]: This is only for the sample or something, right?

File: build.gradle (hunk 1 above)
Comment (NightlyNexus) [reply build.gradle]: yeah, looks like i was.

File: build.gradle (hunk 2 above)
Comment (NightlyNexus) [reply build.gradle]: totally curious: why not?

File: build.gradle (hunk 3 above)
Comment (NightlyNexus) [reply build.gradle]: yeah.

File: build.gradle (hunk 2 above)
Comment (JakeWharton) [reply build.gradle]: I don't want to start inflicting this on people forcefully until it's stable. Granted, we aren't going to have a release right away or anything.

Anyway there's going to be an RC2 any second now and with any luck 1.0 final will be a week after that.

Comment (NightlyNexus) [reply build.gradle]: cool, i was just curious if it was a timebomb thing again or something weird.

Comment (Shamus-Murray-ck) [reply build.gradle]: @JakeWharton any news on the RC2 and/or 1.0 final release?

**Comment by JakeWharton** [comment]
Rebase? 1.0 is releasing soon. It's already a week overdue...

**Comment by NightlyNexus** [comment]
interestingly, the lifecycle stuff is included in androidx core.

File: build.gradle [hunk 4]
```diff
@@ -5,9 +5,10 @@ buildscript {
     'targetSdk': 28,
     'sourceCompatibility': JavaVersion.VERSION_1_7,
     'targetCompatibility': JavaVersion.VERSION_1_7,
+    'androidx': '1.0.0-rc02',
+    'androidxLifecycle': '2.0.0-rc01',
```
Comment (JakeWharton) [inline build.gradle]: These should be inlined. It's only coincidence that they're all at the same version now. Once 1.0 lands they will be versioned individually like normal libraries.

Comment (NightlyNexus) [reply build.gradle]: great!

**Comment by jrodbx** [comment]
@NightlyNexus mind rebasing?

**Comment by jrodbx** [comment]
Thanks!

**Comment by uziassantosferreira** [comment]
should I use this snapshot?

**Comment by kzotin** [comment]
Hi friends,
Any plans to publish update with androidX dependencies? 

We're trying to get rid of **Jetifier** from the build...

**Comment by jrodbx** [comment]
We'll backport the fix to 2.x in order to unblock efforts to rid Jetifier from builds: https://github.com/square/picasso/issues/2175

**Pull request merged**